src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -62,6 +72,14 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
                 return (OracleResponseCode.Error, null);
             if (!Settings.Default.AllowedContentTypes.Contains(message.Content.Headers.ContentType.MediaType))
                 return (OracleResponseCode.ContentTypeNotSupported, null);
+
+            if (!Settings.Default.AllowPrivateHost && message.RequestMessage.RequestUri != uri)
```

Comment:
I don't think it really works to prevent ssrf since the request is already sent in the line:

```
message = await client.GetAsync(uri, HttpCompletionOption.ResponseContentRead, cancellation);
```

Reply:
Yes. It will leak information such as resource existence at least because this check is not the first branch.

Reply:
If we don't want the redirection to occur, disabling it directly may be better.

Or we'd better add a private host check for `message.RequestMessage.RequestUri`.

Reply:
Agree with @vang1ong7ang , I think that the redirection must be manual like https://github.com/neo-project/neo-modules/pull/692

Reply:
yeah, should not assume `GET` requests have no side effect.